src/Application/SqlScripts/SqlTraits.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Ravenfire\Magpie\Application\SqlScripts;
+
+/**
+ * Traits used by the sql scripts
+ */
+trait SqlTraits
+{
+    /**
+     * Limits the width of table columns to 12 characters.
+     *
+     * @param $results
+     * @param $db_columns
+     * @return array
+     */
+    public function setStrLen($results, $db_columns)
+    {
+        foreach ($results[0] as $result => $data) {
+            $db_columns[] = $result;
```

Comment:
I don't think this will work as you expect. `$db_columns` is passed in as an empty array. Arrays are passed by reference **except when they are modified in the function**, then they get copied.

Which means the `$db_columns` outside this function is not modified **by** the function. So $db_columns should still be empty when you want to use it.

The easiest way to fix it is to change the function signature to `setStrLen($results, &$db_columns)` which tells php "let this function update `$db_columns`

https://stackoverflow.com/questions/2030906/are-arrays-in-php-copied-as-value-or-as-reference-to-new-variables-and-when-pas

Reply:
So I ran both the sql:find and the sql:join commands. Both work as expected. When I look at it 'db_columns' is being modified in the function in the first for loop.

Reply:
But do those modification persist after the function? On line 58 of the SqlFind command?

I believe you if you say it does. Just from my understanding, it **shouldn't**. My understanding could be wrong or outdated.

Reply:
I see what you're saying now. You were right. It'll be fixed on the next push.